### PR TITLE
Create pending checks when processing starts

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Adds a hook to the `checks.SuitesAPI` for a pending run for the given product + sha when the taskcluster webhook successfully kicks off the processor for it.

### CODE REVIEW

File: api/webhook/taskcluster.go [hunk 1]
```diff
@@ -152,7 +153,15 @@ func handleStatusEvent(ctx context.Context, payload []byte) (bool, error) {
...
-	err = createAllRuns(log, urlfetch.Client(slowCtx), api, *status.SHA, username, password, urlsByBrowser, labels)
+	suitesAPI := checks.NewSuitesAPI(ctx)
+	err = createAllRuns(log,
+		urlfetch.Client(slowCtx),
+		suitesAPI,
+		uploadURL,
+		*status.SHA,
+		username,
+		password,
+		urlsByBrowser, labels)
```
Comment (Hexcles) [inline api/webhook/taskcluster.go]: Nit: be consistent about the styling; put `labels` into a new line.

File: api/webhook/taskcluster.go [hunk 2]
```diff
@@ -264,9 +274,12 @@ func createAllRuns(log shared.Logger,
 		go func(browser string, urls []string) {
 			defer wg.Done()
 			log.Infof("Reports for %s: %v", browser, urls)
-			err := createRun(client, sha, api, username, password, urls, labels)
+			err := createRun(client, sha, uploadURL, username, password, urls, labels)
 			if err != nil {
 				errors <- err
+			} else {
+				browserName := strings.Split(browser, "-")[0] // chrome-dev => chrome
```
Comment (Hexcles) [inline api/webhook/taskcluster.go]: Why do you want to extract just the browser name and leave the channel aside? We could have chrome-dev & chrome-stable on the same commit.

File: api/webhook/taskcluster.go [hunk 3]
```diff
@@ -264,9 +274,12 @@ func createAllRuns(log shared.Logger,
...
+				suitesAPI.PendingCheckRun(sha, browserName)
```
Comment (Hexcles) [inline api/webhook/taskcluster.go]: Do you have a rough idea how long this API call may take?

Comment (lukebjerring) [reply api/webhook/taskcluster.go]: Anecdotally, requests are quick (mostly <100ms, no total request times for requests that send these are over 1s)

File: api/checks/webhook.go [hunk 4]
```diff
@@ -108,15 +108,17 @@ func handleCheckRunEvent(ctx context.Context, payload []byte) (bool, error) {
...
-	if checkRun.Action != nil &&
-		(*checkRun.Action == "created" || *checkRun.Action == "rerequested") {
-		name, sha := *checkRun.CheckRun.Name, *checkRun.CheckRun.HeadSHA
-		log.Debugf("Check run %s @ %s %s", name, sha[:7], *checkRun.Action)
-		spec, err := shared.ParseProductSpec(*checkRun.CheckRun.Name)
-		if err != nil {
-			log.Errorf("Failed to parse \"%s\" as product spec")
+	if checkRun.Action != nil {
+		if (*checkRun.Action == "created" && *checkRun.CheckRun.Status != "completed") ||
+			*checkRun.Action == "rerequested" {
```
Comment (Hexcles) [inline api/checks/webhook.go]: FWIW, I don't actually know the semantics of these actions/statuses, so I'll trust your investigation.

File: api/checks/webhook.go [hunk 5]
```diff
@@ -108,15 +108,17 @@ func handleCheckRunEvent(ctx context.Context, payload []byte) (bool, error) {
...
+			name, sha := *checkRun.CheckRun.Name, *checkRun.CheckRun.HeadSHA
+			log.Debugf("Check run %s @ %s %s", name, sha[:7], *checkRun.Action)
```
Comment (Hexcles) [inline api/checks/webhook.go]: Nit: let's just log the full hash here.

File: api/checks/webhook.go [hunk 6]
```diff
@@ -108,15 +108,17 @@ func handleCheckRunEvent(ctx context.Context, payload []byte) (bool, error) {
...
 	}
 
-	if checkRun.Action != nil &&
-		(*checkRun.Action == "created" || *checkRun.Action == "rerequested") {
-		name, sha := *checkRun.CheckRun.Name, *checkRun.CheckRun.HeadSHA
-		log.Debugf("Check run %s @ %s %s", name, sha[:7], *checkRun.Action)
-		spec, err := shared.ParseProductSpec(*checkRun.CheckRun.Name)
-		if err != nil {
-			log.Errorf("Failed to parse \"%s\" as product spec")
+	if checkRun.Action != nil {
```
Comment (Hexcles) [inline api/checks/webhook.go]: Nit: we have too many levels of nesting here and this one feels like an error check, so let's convert this top-level if into an early return.

**Pull request merged**